packages/p2p-media-loader-core/src/http-loader.ts

```diff
@@ -50,10 +53,18 @@ export class HttpRequestExecutor {
   private async fetch() {
     const { segment } = this.request;
     try {
-      const response = await window.fetch(segment.url, {
+      const request = new Request(segment.url, {
         headers: this.requestHeaders,
         signal: this.abortController.signal,
       });
+
+      this.settings.httpRequestSetup?.(
+        segment.url,
+        this.byteRange as ByteRange,
```

Comment:
byteRange can be undefined here. Using `as` is usually a hack that breaks types.